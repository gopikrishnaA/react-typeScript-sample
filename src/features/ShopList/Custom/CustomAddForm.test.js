import React from "react";
import { shallow } from "enzyme";
import renderer from 'react-test-renderer';
import { Form, Input } from "reactstrap";
import CustomAddForm from "./CustomAddForm";

describe("CustomAddForm", () => {
    let wrapper;
    let useEffect;

    let props = {
        isShown: false,
        onAdd: jest.fn(),
        onClose: jest.fn(),
    };

    const mockUseEffect = () => {
        useEffect.mockImplementationOnce(f => f());
    };

    beforeEach(() => {
        useEffect = jest.spyOn(React, "useEffect");
        mockUseEffect();

        wrapper = shallow(<CustomAddForm {...props} />);
    });

    it("renders correctly", () => {
        const tree = renderer
            .create(wrapper.find(Form))
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("should setName of the input", () => {
        const event =  { target: { value: 'Hello' } };
        const nameInput = wrapper.find(Input).at(0);
        nameInput.simulate('change', event);
        expect(nameInput.props().value).toBe('');
    });

    it("should setAmount of the input", () => {
        const event =  { target: { value: '1000' } };
        const nameInput = wrapper.find(Input).at(1);
        nameInput.simulate('change', event);
        expect(nameInput.props().value).toBe(0);
    });

    it("should setUnit of the input", () => {
        const event =  { target: { value: '1' } };
        const nameInput = wrapper.find(Input).at(2);
        nameInput.simulate('change', event);
        expect(nameInput.props().value).toBe('');
    });
    it("should click on submit of the form", () => {
        const event =  { 
            preventDefault: jest.fn(), 
            target: { value: '1' } };
        wrapper.find(Form).props().onSubmit(event);
        expect(props.onAdd).toHaveBeenCalledTimes(1);
    });
});
