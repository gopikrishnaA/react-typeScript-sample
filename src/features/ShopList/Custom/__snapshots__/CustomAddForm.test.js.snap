// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomAddForm renders correctly 1`] = `
<form
  className=""
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <label
      className=""
      htmlFor="name"
    >
      Name
    </label>
    <input
      className="form-control"
      name="name"
      onChange={[Function]}
      required={true}
      type="text"
      value=""
    />
  </div>
  <div
    className="form-group"
  >
    <label
      className=""
      htmlFor="name"
    >
      Amount
    </label>
    <input
      className="form-control"
      name="amount"
      onChange={[Function]}
      type="text"
      value={0}
    />
  </div>
  <div
    className="form-group"
  >
    <label
      className=""
      htmlFor="name"
    >
      Unit
    </label>
    <input
      className="form-control"
      name="unit"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <button
    aria-label={null}
    className="btn btn-success"
    onClick={[Function]}
    type="submit"
  >
    Add
  </button>
  <button
    aria-label={null}
    className="ml-2 btn btn-danger"
    onClick={[Function]}
    type="button"
  >
    Cancel
  </button>
</form>
`;
